compiler/codegen/OMRCodeGenerator.hpp

```diff
@@ -489,8 +489,30 @@ class OMR_EXTENSIBLE CodeGenerator
    bool isRegisterClobberable(TR::Register *reg, uint16_t count);
 
    // ilgen
-   bool ilOpCodeIsSupported(TR::ILOpCodes); // no virt, default, cast
-
+   /**
+    * @brief Returns if an IL OpCode is supported by current CodeGen
+    *
+    * @param op The IL OpCode being checked.
+    *
+    * @return True if the IL OpCode is supported otherwise false.
+    */
+   static bool ilOpCodeIsSupported(TR::ILOpCodes op); // no virt, default, cast
```

Comment:
Isn't the whole point of OMR that all IL be platform agnostic? Why do we need this query exactly?

Reply:
Each platform only supports a subset of OMR IL OpCode; therefore, the Optimizer need know whether an OpCode is supported before utilizing it. One example is that Power supports many math OpCode while X86 doesn't, i.e. `dsin`, `dcos`, `dlog`, etc.

Reply:
Thanks for the explanation.